projects/wave-core/src/lib/util/pipes/highlight.pipe.ts

```diff
@@ -1,22 +1,19 @@
-import {Pipe, PipeTransform} from '@angular/core';
+import { PipeTransform, Pipe } from '@angular/core';
 
 @Pipe({name: 'waveHighlightPipe'})
-export class HighlightPipe implements PipeTransform {
 
-    // TODO: replace <span> with a <higlight> component?
-    transform(text: string, term: string): string {
+export class HighlightPipe implements PipeTransform {
+    transform(text: string, search): string {
+        if (search && text) {
+            let pattern = search.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, '\\$&');
+            pattern = pattern.split(' ').filter((t) => {
+                return t.length > 0;
+            }).join('|');
```

Comment:
What do you accomplish in comparison to the previous functionality by replacing something on the search term and by splitting it into several parts?

Reply:
changed back to old regex